identity/aziot-identityd/src/lib.rs

```diff
@@ -825,8 +865,8 @@ fn create_csr(
     subject_name.append_entry_by_nid(openssl::nid::Nid::COMMONNAME, subject)?;
     let subject_name = subject_name.build();
     csr.set_subject_name(&subject_name)?;
-    csr.set_pubkey(public_key)?;
-    csr.sign(private_key, openssl::hash::MessageDigest::sha256())?;
+    csr.set_pubkey(key)?;
```

Comment:
The private key shouldn't be set as the public key in the CSR. Please revert this.

Reply:
This was something I wasn't sure about, but it since openssl::pkey::Private implements the HasPublic trait, it should be fine to pass the private key to set_pubkey, and that function will generate the public key from the private key. Is that not how this works?

Reply:
It depends on the algorithm and exponent used. For provisioning, identity cert key openssl pkeyrefs are obtained using the key engine and we cannot assume anything about the properties of the key. 

---
In reply to: [513767139](https://github.com/Azure/iot-identity-service/pull/61#discussion_r513767139) [](ancestors = 513767139)

Reply:
Got it. Fixed.